native/src/main/java/io/ballerina/lib/ibm.ibmmq/CommonUtils.java

```diff
@@ -18,23 +18,91 @@
 
 package io.ballerina.lib.ibm.ibmmq;
 
+import com.ibm.mq.MQException;
+import com.ibm.mq.MQMessage;
 import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
 import io.ballerina.runtime.api.utils.StringUtils;
 import io.ballerina.runtime.api.values.BError;
 import io.ballerina.runtime.api.values.BMap;
 import io.ballerina.runtime.api.values.BString;
 
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Enumeration;
 import java.util.Optional;
 
+import static io.ballerina.lib.ibm.ibmmq.Constants.IBMMQ_ERROR;
+import static io.ballerina.lib.ibm.ibmmq.ModuleUtils.getModule;
+
 /**
  * {@code CommonUtils} contains the common utility functions for the Ballerina IBM MQ connector.
  */
 public class CommonUtils {
 
+    public static final String BTOPIC = "TOPIC";
+    public static final String TOPIC_OBJECT = "TOPIC_OBJECT";
+
+    private static final String ERROR_DETAILS = "ErrorDetails";
+    private static final BString ERROR_REASON_CODE = StringUtils.fromString("ErrorDetails");
+    private static final BString ERROR_ERROR_CODE = StringUtils.fromString("ErrorDetails");
+    private static final BString ERROR_COMPLETION_CODE = StringUtils.fromString("ErrorDetails");
+    private static final BString MESSAGE_PAYLOAD = StringUtils.fromString("payload");
+    private static final BString MESSAGE_PROPERTIES = StringUtils.fromString("properties");
+    private static final String BMESSAGE_NAME = "Message";
+
+    public static MQMessage getMqMessageFromMessage(BMap<BString, Object> bMessage) {
+        byte[] payload = bMessage.getArrayValue(MESSAGE_PAYLOAD).getBytes();
+        BMap<BString, Object> properties = (BMap<BString, Object>) bMessage.getMapValue(MESSAGE_PROPERTIES);
+
+        MQMessage mqMessage = new MQMessage();
+        try {
+            mqMessage.write(payload);
+        } catch (IOException e) {
+            throw createError(IBMMQ_ERROR,
+                    String.format("Error occurred while populating payload: %s", e.getMessage()), e);
+        }
+        for (BString key : properties.getKeys()) {
```

Comment:
Lets set the properties based on the relevant type rather than `Object` properties.

Reply:
Updated